docs/en/tutorials/getting-started.rst

```diff
@@ -228,39 +227,235 @@ entity definition:
+entity.
+
+.. note::
+
+    A `DTO <https://en.wikipedia.org/wiki/Data_transfer_object>` is an object
+    that only carries data without any logic. Its only goal is to be transferred
+    from one service to another.
+    A ``DTO`` often represents data sent by a client and that has to be validated,
+    but can also be used as simple data carrier for other cases.
+
+By using ``DTOs``, if we take our previous ``User`` example, we could create
+a ``ProfileEditingForm`` DTO that will be a plain model, totally unrelated to
+our database, that will be populated via a form and validated.
+Then we can add a new mutator to our ``User``:
+
+.. configuration-block::
+
+    .. code-block:: php
+        <?php
+        class User
         {
-            $this->name = $name;
+            public function updateFromProfile(ProfileEditingForm $profileForm): void
+            {
+                // ...
+            }
+
+            public static function createFromRegistration(UserRegistrationForm $registrationForm): self
+            {
+                // ...
+            }
         }
-    }
 
-When creating entity classes, all of the fields should be protected or private
-(not public), with getter and setter methods for each one (except $id).
-The use of mutators allows Doctrine to hook into calls which
-manipulate the entities in ways that it could not if you just
-directly set the values with ``entity#field = foo;``
-
-The id field has no setter since, generally speaking, your code
-should not set this value since it represents a database id value.
-(Note that Doctrine itself can still set the value using the
-Reflection API instead of a defined setter function)
-
-The next step for persistence with Doctrine is to describe the
-structure of the ``Product`` entity to Doctrine using a metadata
-language. The metadata language describes how entities, their
-properties and references should be persisted and what constraints
-should be applied to them.
+There are several advantages to using such a model:
+
+* **Entity state is always valid**. Since no setters exist, this means that we
+only update portions of the entity that should already be valid.
+* Instead of having plain getters and setters, our entity now has
+**real behavior**: it is much easier to determine the logic in the domain.
+* DTOs can be reused in other components, for example deserializing mixed
+content, using forms...
+* Classic and static constructors can be used to manage different ways to
+create our objects, and they can also use DTOs.
+* Anemic entities tend to isolate the entity from logic, whereas rich
+entities allow putting the logic in the object itself, including data
+validation.
+
+The next step for persistence with Doctrine is to describe the structure of
+the ``Product`` entity to Doctrine using a metadata language. The metadata
+language describes how entities, their properties and references should be
+persisted and what constraints should be applied to them.
```

Comment:
The section following this edit mentions again setters and getters. That's IMO confusing.